[2024-12-22T14:41:19.248+0000] {logging_mixin.py:149} WARNING - /home/***/.local/lib/python3.7/site-packages/***/utils/sqlalchemy.py:124 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2024-12-22T14:41:19.255+0000] {taskinstance.py:1125} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: flight_search_dag.flight_search_ingestion scheduled__2024-12-22T13:00:00+00:00 [queued]>
[2024-12-22T14:41:19.260+0000] {taskinstance.py:1125} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: flight_search_dag.flight_search_ingestion scheduled__2024-12-22T13:00:00+00:00 [queued]>
[2024-12-22T14:41:19.260+0000] {taskinstance.py:1331} INFO - Starting attempt 3 of 3
[2024-12-22T14:41:19.267+0000] {taskinstance.py:1350} INFO - Executing <Task(SparkSubmitOperator): flight_search_ingestion> on 2024-12-22 13:00:00+00:00
[2024-12-22T14:41:19.269+0000] {standard_task_runner.py:57} INFO - Started process 775 to run task
[2024-12-22T14:41:19.270+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'flight_search_dag', 'flight_search_ingestion', 'scheduled__2024-12-22T13:00:00+00:00', '--job-id', '20', '--raw', '--subdir', 'DAGS_FOLDER/flight_search_dag.py', '--cfg-path', '/tmp/tmpaqaofhs7']
[2024-12-22T14:41:19.271+0000] {standard_task_runner.py:85} INFO - Job 20: Subtask flight_search_ingestion
[2024-12-22T14:41:19.279+0000] {logging_mixin.py:149} WARNING - /home/***/.local/lib/python3.7/site-packages/***/settings.py:188 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2024-12-22T14:41:19.306+0000] {task_command.py:410} INFO - Running <TaskInstance: flight_search_dag.flight_search_ingestion scheduled__2024-12-22T13:00:00+00:00 [running]> on host c542defac8ed
[2024-12-22T14:41:19.314+0000] {logging_mixin.py:149} WARNING - /home/***/.local/lib/python3.7/site-packages/***/utils/sqlalchemy.py:124 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2024-12-22T14:41:19.355+0000] {taskinstance.py:1570} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='flight_search_dag' AIRFLOW_CTX_TASK_ID='flight_search_ingestion' AIRFLOW_CTX_EXECUTION_DATE='2024-12-22T13:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-22T13:00:00+00:00'
[2024-12-22T14:41:19.360+0000] {base.py:73} INFO - Using connection ID 'spark_default' for task execution.
[2024-12-22T14:41:19.361+0000] {spark_submit.py:339} INFO - Spark-Submit cmd: spark-submit --master spark://spark-container:7077 --total-executor-cores 2 --executor-cores 1 --executor-memory 1g --driver-memory 1g --name arrow-spark --verbose --queue default /shared_volume/data_ingestion.py
[2024-12-22T14:41:19.402+0000] {spark_submit.py:490} INFO - /home/***/.local/lib/python3.7/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2024-12-22T14:41:19.402+0000] {spark_submit.py:490} INFO - /home/***/.local/lib/python3.7/site-packages/pyspark/bin/spark-class: line 71: /usr/lib/jvm/java-17-openjdk-amd64/bin/java: No such file or directory
[2024-12-22T14:41:19.403+0000] {spark_submit.py:490} INFO - /home/***/.local/lib/python3.7/site-packages/pyspark/bin/spark-class: line 97: CMD: bad array subscript
[2024-12-22T14:41:19.410+0000] {taskinstance.py:1847} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-container:7077 --total-executor-cores 2 --executor-cores 1 --executor-memory 1g --driver-memory 1g --name arrow-spark --verbose --queue default /shared_volume/data_ingestion.py. Error code is: 1.
[2024-12-22T14:41:19.411+0000] {taskinstance.py:1373} INFO - Marking task as FAILED. dag_id=flight_search_dag, task_id=flight_search_ingestion, execution_date=20241222T130000, start_date=20241222T144119, end_date=20241222T144119
[2024-12-22T14:41:19.417+0000] {standard_task_runner.py:109} ERROR - Failed to execute job 20 for task flight_search_ingestion (Cannot execute: spark-submit --master spark://spark-container:7077 --total-executor-cores 2 --executor-cores 1 --executor-memory 1g --driver-memory 1g --name arrow-spark --verbose --queue default /shared_volume/data_ingestion.py. Error code is: 1.; 775)
[2024-12-22T14:41:19.454+0000] {local_task_job_runner.py:232} INFO - Task exited with return code 1
[2024-12-22T14:41:19.467+0000] {taskinstance.py:2674} INFO - 0 downstream tasks scheduled from follow-on schedule check
